include/nbl/builtin/hlsl/spirv_intrinsics/core.hlsl

```diff
@@ -47,199 +69,1272 @@ static const uint32_t3 LocalInvocationId;
+
+[[vk::ext_instruction(spv::OpAtomicUMax)]]
+uint32_t atomicMax([[vk::ext_reference]] uint32_t pointer, uint32_t memoryScope,  uint32_t semantics, uint32_t value);
+
+[[vk::ext_instruction(spv::OpAtomicUMax)]]
+uint64_t atomicMax([[vk::ext_reference]] uint64_t pointer, uint32_t memoryScope,  uint32_t semantics, uint64_t value);
+
+template<typename P>
+[[vk::ext_instruction(spv::OpAtomicUMax)]]
+enable_if_t<is_spirv_type_v<P>, uint32_t> atomicMax(P pointer, uint32_t memoryScope,  uint32_t semantics, uint32_t value);
+
+template<typename P>
+[[vk::ext_instruction(spv::OpAtomicUMax)]]
+enable_if_t<is_spirv_type_v<P>, uint64_t> atomicMax(P pointer, uint32_t memoryScope,  uint32_t semantics, uint64_t value);
 
 template<typename T>
 [[vk::ext_instruction(spv::OpAtomicAnd)]]
-enable_if_t<is_same_v<T,uint32_t> || is_same_v<T,int32_t>, T> atomicAnd([[vk::ext_reference]] T ptr, uint32_t memoryScope, uint32_t memorySemantics, T value);
+T atomicAnd([[vk::ext_reference]] T pointer, uint32_t memoryScope,  uint32_t semantics, T value);
 
-template<typename T, typename Ptr_T> // DXC Workaround
+template<typename T, typename P>
 [[vk::ext_instruction(spv::OpAtomicAnd)]]
-enable_if_t<is_spirv_type_v<Ptr_T> && (is_same_v<T,uint32_t> || is_same_v<T,int32_t>), T> atomicAnd(Ptr_T ptr, uint32_t memoryScope, uint32_t memorySemantics, T value);
+enable_if_t<is_spirv_type_v<P>, T> atomicAnd(P pointer, uint32_t memoryScope,  uint32_t semantics, T value);
 
 template<typename T>
 [[vk::ext_instruction(spv::OpAtomicOr)]]
-enable_if_t<is_same_v<T,uint32_t> || is_same_v<T,int32_t>, T> atomicOr([[vk::ext_reference]] T ptr, uint32_t memoryScope, uint32_t memorySemantics, T value);
+T atomicOr([[vk::ext_reference]] T pointer, uint32_t memoryScope,  uint32_t semantics, T value);
 
-template<typename T, typename Ptr_T> // DXC Workaround
+template<typename T, typename P>
 [[vk::ext_instruction(spv::OpAtomicOr)]]
-enable_if_t<is_spirv_type_v<Ptr_T> && (is_same_v<T,uint32_t> || is_same_v<T,int32_t>), T> atomicOr(Ptr_T ptr, uint32_t memoryScope, uint32_t memorySemantics, T value);
+enable_if_t<is_spirv_type_v<P>, T> atomicOr(P pointer, uint32_t memoryScope,  uint32_t semantics, T value);
 
 template<typename T>
 [[vk::ext_instruction(spv::OpAtomicXor)]]
-enable_if_t<is_same_v<T,uint32_t> || is_same_v<T,int32_t>, T> atomicXor([[vk::ext_reference]] T ptr, uint32_t memoryScope, uint32_t memorySemantics, T value);
+T atomicXor([[vk::ext_reference]] T pointer, uint32_t memoryScope,  uint32_t semantics, T value);
 
-template<typename T, typename Ptr_T> // DXC Workaround
+template<typename T, typename P>
 [[vk::ext_instruction(spv::OpAtomicXor)]]
-enable_if_t<is_spirv_type_v<Ptr_T> && (is_same_v<T,uint32_t> || is_same_v<T,int32_t>), T> atomicXor(Ptr_T ptr, uint32_t memoryScope, uint32_t memorySemantics, T value);
+enable_if_t<is_spirv_type_v<P>, T> atomicXor(P pointer, uint32_t memoryScope,  uint32_t semantics, T value);
```

Comment:
same as with the compare or swap atomics, caps/extensions for sizes other than 32bit